src/index.test.ts

```diff
@@ -1,9 +1,121 @@
 import { expect, test, vi } from "vitest";
 import { main } from "./index";
+import {
+  boardStateToString,
+  checkBoardStateConsistency,
+  GameState,
+  printBoardStateToConsole,
+  runConnect4,
+} from "./connect4";
+interface TestingGameState extends GameState {
+  boardDisplay?: String;
+}
 
-test("main function", () => {
+const correctGameState: TestingGameState = {
+  boardState: [
+    [0, 0, 0, 0, 1, 0, 0],
+    [0, 0, 0, 0, 2, 0, 0],
+    [0, 2, 1, 0, 2, 0, 0],
+    [0, 2, 1, 1, 2, 0, 2],
+    [0, 1, 1, 2, 1, 1, 1],
+    [0, 2, 2, 1, 1, 2, 2],
+  ],
+  boardDisplay: `
++---+---+---+---+---+---+---+
+|   |   |   |   | o |   |   |
+|   |   |   |   | x |   |   |
+|   | x | o |   | x |   |   |
+|   | x | o | o | x |   | x |
+|   | o | o | x | o | o | o |
+|   | x | x | o | o | x | x |
++---+---+---+---+---+---+---+
+  1   2   3   4   5   6   7  
+`,
+};
+
+const flyingTokenIncorrectGameState1: TestingGameState = {
+  boardState: [
+    [1, 0, 0, 0, 1, 0, 0],
+    [0, 0, 0, 0, 2, 0, 0],
+    [0, 2, 1, 0, 2, 0, 0],
+    [0, 2, 1, 1, 2, 0, 2],
+    [0, 1, 1, 2, 1, 1, 1],
+    [0, 2, 2, 1, 1, 2, 2],
+  ],
+};
+
+const flyingTokenIncorrectGameState2: TestingGameState = {
+  boardState: [
+    [0, 0, 0, 0, 1, 0, 0],
+    [0, 0, 0, 0, 2, 0, 0],
+    [0, 2, 1, 0, 2, 1, 0],
+    [0, 2, 1, 1, 2, 0, 2],
+    [0, 1, 1, 2, 1, 1, 1],
+    [0, 2, 2, 1, 1, 2, 2],
+  ],
+};
+
+const nbTokenIncorrectGameState1: TestingGameState = {
+  boardState: [
+    [0, 0, 0, 0, 1, 0, 0],
+    [0, 0, 0, 0, 2, 0, 0],
+    [0, 2, 1, 0, 2, 0, 0],
+    [0, 2, 1, 1, 2, 0, 2],
+    [1, 1, 1, 2, 1, 1, 1],
+    [1, 2, 2, 1, 1, 2, 2],
+  ],
+};
+
+const nbTokenIncorrectGameState2: TestingGameState = {
+  boardState: [
+    [0, 0, 0, 0, 1, 0, 0],
+    [0, 0, 0, 0, 2, 0, 0],
+    [0, 2, 1, 0, 2, 2, 0],
+    [0, 2, 1, 1, 2, 2, 2],
+    [0, 1, 1, 2, 1, 1, 1],
+    [0, 2, 2, 1, 1, 2, 2],
+  ],
+};
+
+test("boardStateToString", () => {
+  expect(boardStateToString(correctGameState.boardState)).toBe(
+    correctGameState.boardDisplay,
+  );
+});
+
+test("printBoardStateToConsole", () => {
+  const consoleSpy = vi.spyOn(console, "log");
+  const testingString: string = "Test console display";
+  printBoardStateToConsole(testingString);
+  expect(consoleSpy).toHaveBeenCalledWith(testingString);
+});
+
+test("checkBoardStateConsistency flying token 1", () => {
+  expect(() =>
+    checkBoardStateConsistency(flyingTokenIncorrectGameState1.boardState),
+  ).toThrowError("Given game state text contains missplaced token(s)");
+});
+
+test("checkBoardStateConsistency flying token 2", () => {
```

Comment:
```suggestion
test("checkBoardStateConsistency flying token 2", () => {
```

I think we can improve these tests names so we can understand what we expect just by reading the name, first by using the describe as discussed above, second by using a complete sentence to describe the test, for example:
`It should throws an error if a token is flying`